### PR TITLE
Fix leaderboard office filter

### LINKED ISSUE #8: Fix the leaderboard office filter
The leaderboard query only returns the top 10 results for the current scope. The office filter only operates on the array client side so when filtering by office the result set is incomplete. When the office filter is changed, we should issue a new fetch for that office and let the apollo store handle caching it accordingly.

### PR DESCRIPTION
**Pull request by blakegong**
Fixes #8.

### CODE REVIEW

**Review by roseleaf — APPROVED** [review]

**Review by ragurney — APPROVED** [review]

**Comment by blakegong** [comment]
@zendesk/volunteer please let me know how I can trigger the CI to be run (otherwise as non-admin with pending CI, the merge button is unavailable).

Or feel free to help me merge this if more time is needed to fix CI. Thank you!

**Pull request merged**
